site/components/form/DateSelector.tsx

```diff
@@ -76,17 +78,36 @@ const DateSelector = <T extends object>({
     optional = false,
     disablePast,
     stateUpdater,
+    reset = false,
+    showError = false,
 }: DateSelectorProps<T>): ReactElement => {
     const [dateValue, setDateValue] = useState<Date | null>(!!disabled || !value ? null : new Date(value));
     const [errors, setErrors] = useState<ErrorInfo[]>(initialErrors);
 
+    useEffect(() => {
+        if (reset) {
+            setDateValue(null);
+        }
+    }, [reset]);
```

Comment:
It looks like reset comes to inputBox as a boolean, which we default to false, and it never changes (because we dont use state for it). As this is the case, I don't think we need a useEffect because that will only happen on first render, and whenever reset changes (as its a dependency - but it wont ever change as its not stateful).

I think it would make more sense to have it in this line:

`const [dateValue, setDateValue] = useState<Date | null>(!!disabled || !value || !!reset ? null : new Date(value));`

Also, regarding the `new Date(value)` part, we have a getDate function we use dayJs inside of, would that be more appropriate here?